pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -593,20 +595,59 @@ func parseAppInstanceConfig(config *zconfig.EdgeDevConfig,
 
 		appInstance.CloudInitUserData = userData
 		appInstance.RemoteConsole = cfgApp.GetRemoteConsole()
+		appInstanceList = append(appInstanceList, appInstance)
+	}
+
+	// validate app instances configuration
+	validateAppInstanceConfig(appInstanceList)
+
+	// publish the config objects
+	for _, appInstance := range appInstanceList {
+		uuidStr := appInstance.UUIDandVersion.UUID.String()
+		// write to zedmanager config directory
+		publishAppInstanceConfig(getconfigCtx, *appInstance)
 		// get the certs for image sha verification
 		certInstance := getCertObjects(appInstance.UUIDandVersion,
 			appInstance.ConfigSha256, appInstance.StorageConfigList)
-
-		// write to zedmanager config directory
-		uuidStr := cfgApp.Uuidandversion.Uuid
-		publishAppInstanceConfig(getconfigCtx, appInstance)
 		if certInstance != nil {
 			publishCertObjConfig(getconfigCtx, certInstance,
 				uuidStr)
 		}
 	}
 }
 
+func validateAppInstanceConfig(appInstanceList []*types.AppInstanceConfig) {
+	log.Infof("check for duplicate port map acls")
+	idx0 := 0
+	// For app instances, attached to the same network instance,
+	//  check for common port map rules
+	for _, appInstance0 := range appInstanceList {
+		idx1 := 0
+		ulCfgList0 := appInstance0.UnderlayNetworkList
+		for _, appInstance1 := range appInstanceList {
```

Comment:
These loops might be more efficient if array indexing is used instead of range. That way you can also avoid the check for "idx1 > idx0".

Reply:
sure